File: src/main/java/io/github/bot/terminal/application/domains/common/notify/entity/Notify.java
package io.github.bot.terminal.application.domains.common.notify.entity;


import io.github.bot.terminal.application.domains.notificarion_api.entity.Message;
import io.github.bot.terminal.application.domains.notificarion_api.entity.NotificationApi;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import lombok.experimental.Accessors;

@Data
@Accessors(chain = true)
@RequiredArgsConstructor
public class Notify {
    private final NotificationApi notificationApi;
    private final String messageTemplate;

    public void execute() {
        notificationApi.sendMessage(new Message(messageTemplate));
    }
}

File: src/main/kotlin/io/github/bot/terminal/application/domains/common/notify/entity/Notify.kt
package io.github.bot.terminal.application.domains.common.notify.entity

import io.github.bot.terminal.application.domains.notificarion_api.entity.Message
import io.github.bot.terminal.application.domains.notificarion_api.entity.NotificationApi

class Notify(
        val notificationApi: NotificationApi,
        val messageTemplate: String
) {
    fun execute() {
        notificationApi.sendMessage(Message(messageTemplate))
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Notify

        if (notificationApi != other.notificationApi) return false
        if (messageTemplate != other.messageTemplate) return false

        return true
    }

    override fun hashCode(): Int {
        var result = notificationApi.hashCode()
        result = 31 * result + messageTemplate.hashCode()
        return result
    }

}